### DISCUSSION TITLE
How to obtain the Pareto front via NSGA-II

### DISCUSSION THREAD

**Discussion by Zcaic** [mdolab/pyoptsparse#401] [General]
I have a multi-objective problem and I found that when I optimize with NSGA-II, I get only one solution, how can I get the whole one Pareto front? thanks!!!

**Comment by ewu63** [comment]
This information is provided in the files written out by NSGA2.
1. Make sure `PrintOut` is set appropriately (say 1)
2. The information is encoded in `nsga2_best_pop.out` but several other files are available with other information you may find useful.

Sample snippet for the example script `nsga2_multi_objective.py`:
```
import matplotlib.pyplot as plt
import numpy as np
x = np.loadtxt("nsga2_best_pop.out")
plt.plot(x[:,0], x[:,1], '.')
plt.show()
```

This plots the Pareto front for this simple 2-objective problem.
![image](https://github.com/mdolab/pyoptsparse/assets/602725/92eabed2-723e-4f50-ad1a-e35132bf4db0)
